dbms/programs/server/Server.cpp

```diff
@@ -438,6 +438,13 @@ int Server::main(const std::vector<std::string> & /*args*/)
             buildLoggers(*config, logger());
             global_context->setClustersConfig(config);
             global_context->setMacros(std::make_unique<Macros>(*config, "macros"));
+
+            /// Setup protection to avoid accidental DROP for big tables (that are greater than 50 GB by default)
+            if (config->has("max_table_size_to_drop"))
+                global_context->setMaxTableSizeToDrop(config->getUInt64("max_table_size_to_drop"));
```

Comment:
`max_table_size_to_drop` in `Context::setMaxTableSizeToDrop(...)` and `Context::checkTableCanBeDropped(...)` can be accessed from different threads, but it's not atomic. The same for `max_partition_size_to_drop`